src/derivative_utils.jl

```diff
@@ -521,7 +486,10 @@ function build_J_W(alg,u,uprev,p,t,dt,f,uEltypeNoUnits,::Val{true})
       return nothing, nothing
```

Comment:
why would we treat Newton completely differently in this function? The same linearity rules and all of that would apply to split Rosenbrock. This seems completely unnecessary.